trade-finance-manager-api/src/v1/repo/user.repo.upsert-user.test.ts

```diff
@@ -0,0 +1,103 @@
+import { AuditDatabaseRecord, AuditDetails, aValidUserUpsertRequest, MONGO_DB_COLLECTIONS, UserUpsertRequest } from '@ukef/dtfs2-common';
+import { generateAuditDatabaseRecordFromAuditDetails, generateSystemAuditDetails } from '@ukef/dtfs2-common/change-stream';
+import { mongoDbClient } from '../../drivers/db-client';
+import { UserRepo } from './user.repo';
+
+let findOneAndUpdateMock = jest.fn();
+let getCollectionMock = jest.fn();
+
+jest.mock('@ukef/dtfs2-common/change-stream', () => ({
+  generateAuditDatabaseRecordFromAuditDetails: jest.fn(),
+}));
+
+describe('user repo', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+    getCollectionMock.mockResolvedValue({
+      updateOne: findOneAndUpdateMock,
+    });
+  });
+
+  describe('upsertUser', () => {
+    let userUpsertRequest: UserUpsertRequest;
+    let auditDetails: AuditDetails;
+    let emailsOfUserToUpsert: string[];
+
+    const mockAuditRecordResponse = 'example audit record';
+
+    beforeEach(() => {
+      userUpsertRequest = aValidUserUpsertRequest();
+      auditDetails = generateSystemAuditDetails();
+      emailsOfUserToUpsert = ['an-email', 'another-email'];
+    });
+
+    describe('when the update is successful', () => {
+      beforeEach(() => {
+        mockSuccessfulGenerateAuditDatabaseRecordFromAuditDetails();
+        mockSuccessfulUpdateOneResponse();
+      });
+
+      it('calls the correct collection', async () => {
+        mockSuccessfulGenerateAuditDatabaseRecordFromAuditDetails();
+        mockSuccessfulUpdateOneResponse();
+
+        await makeRequest();
+
+        expect(getCollectionMock).toHaveBeenCalledWith(MONGO_DB_COLLECTIONS.TFM_USERS);
```

Comment:
Can we add a `toHaveBeenCalledTimes` assertion please

Reply:
Opps this test file shouldnt exist anymore -- but yes, will add to others